blueman/main/Tray.py

```diff
@@ -1,25 +1,36 @@
 from importlib import import_module
 import logging
-
-from blueman.Functions import check_single_instance
+import os
+import sys
 from blueman.main.DBusProxies import AppletService
-
-import gi
-gi.require_version('Gtk', '3.0')
-from gi.repository import GLib, Gio
+from gi.repository import Gio
 
 
-class BluemanTray:
+class BluemanTray(Gio.Application):
     def __init__(self):
-        check_single_instance("blueman-tray")
+        super().__init__(application_id="org.blueman.Tray", flags=Gio.ApplicationFlags.FLAGS_NONE)
+        self._active = False
 
-        applet = AppletService()
+    def do_startup(self):
+        Gio.Application.do_startup(self)
+
+        quit_action = Gio.SimpleAction.new("Quit", None)
+        quit_action.connect("activate", self.quit)
+        self.add_action(quit_action)
```

Comment:
Is this actually used?

Reply:
Yes in the ManagerService dbusproxy

Reply:
Just stumbled upon this again. I do not seem to get it. I think manager and tray both define a `Quit` action but only the one of manager is used from `ManagerService.startstop`. The tray one's seems to be unused and I even think it cannot work as `self.quit` does not accept the arguments that the `activate` signal expects. :thinking: